src/Upscale.cpp

```diff
@@ -183,7 +183,7 @@ void Upscale::SolveAtLevel(int level, const mfem::Vector& x, mfem::Vector& y) co
 
 mfem::Vector Upscale::SolveAtLevel(int level, const mfem::Vector& x) const
 {
-    mfem::Vector coarse_vect = GetVector(1);
+    mfem::Vector coarse_vect = GetVector(level);
```

Comment:
Good catch! How did this function not break before? Maybe this is never called for `level !=1`.